src/scripts/bootstrap_trigger/bootstrap_trigger.py

```diff
@@ -28,30 +31,29 @@ def read_args():
                               "this data. If querying a --file-view, this is "
                               "usually the scope of that file view (or the "
                               "Synapse ID of a folder which contains everything "
-                              "in the scope). If data is instead coming from the "
-                              "folder specified by --synapse-parent, these values "
-                              "are identical."))
+                              "in the scope)."))
     parser.add_argument("--query",
-                        help=("An f-string formatted query which filters the "
+                        help=("Optional. An f-string formatted query which filters the "
                               "file view. Use {source_table} in the FROM clause. "
                               "If this argument is not specified, all files in "
                               "the --file-view will be submitted to the "
                               "--glue-workflow."))
-    parser.add_argument("--glue-workflow",
-                        required=True,
-                        help="The name of the Glue workflow to submit to.")
     parser.add_argument("--diff-s3-uri",
-                        help=("The S3 URI of a parquet dataset to diff with "
+                        help=("Optional. The S3 URI of a parquet dataset to diff with "
                               "before submitting to the --glue-workflow."))
     parser.add_argument("--diff-parquet-field",
-                        help=("The field name in the parquet dataset to diff "
-                              "upon."))
+                        help=("Optional. The field name in the parquet dataset to diff "
+                              "upon. Defaults to 'recordid'."),
+                        default="recordid")
     parser.add_argument("--diff-file-view-field",
-                        help=("The field name in the --file-view to diff upon."))
+                        help=("Optional. The field name in the --file-view to diff "
+                              "upon. Defaults to 'recordId'."),
+                        default="recordId")
     parser.add_argument("--profile",
-                        help="The AWS profile to use.")
+                        help=("Optional. The AWS profile to use. Uses the default "
+                              "profile if not specified."))
     parser.add_argument("--ssm-parameter",
-                        help=("The name of the SSM parameter containing "
+                        help=("Optional. The name of the SSM parameter containing "
```

Comment:
Nit: https://stackoverflow.com/questions/24180527/argparse-required-arguments-listed-under-optional-arguments.  This isn't necessary, but just some food for thought - Positional arguments are usually "required" arguments vs the argparse default "optional arguments" when you use the `-h` flag.

Reply:
tbh I have to disagree with argparse's reasoning. Why would they list flagged arguments with required=True under "optional arguments" just because there is a convention that flagged arguments be optional? I prefer flagged arguments to positional since they make explicit what is being set. With 3+ required positional arguments things start to get confusing.

Reply:
I agree with you.  I feel that if there is a solution that is provided via that stackoverflow, that they could easily create that argparse group for "required" named parameters, instead of having users create their own group.